### DISCUSSION TITLE
Add `scoop` or `winget` for installation under windows platform

### DISCUSSION THREAD

**Discussion by rgbygv** [libnyanpasu/clash-nyanpasu#385] [Ideas]
Use `scoop` or `winget` can simplify installation and updates.

```
PS C:\Users\xxx> scoop search clash
clash/main 1.18.0
clash-verge/extras 1.3.8
clash-verge-rev/dorado 1.5.0
clash-verge-rev/extras 1.5.0 
clash.mini/extras 0.2.2
clashtui/extras 0.1.0
PS C:\Users\xxx> winget search clash
Name                     Id                        Version Match      Source
----------------------------------------------------------------------------
clash                    Dreamacro.clash           1.18.0             winget
Xray-core                XTLS.Xray-core            1.7.5   Tag: clash winget
sing-box                 SagerNet.sing-box         1.8.4   Tag: clash winget
Clash Verge              GyDi.ClashVerge           1.3.8   Tag: clash winget
Corporate Clash Launcher CorporateClashCrew.Toont… 1.3.0              winget
```

**Comment by greenhat616** [comment]
我不太了解这个。能否提供下文档告知一下我们怎么去给 scoop、winget 添加新软体呢？

**Reply by 66Leo66** [reply]
https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md#for-scoop-buckets

**Comment by rgbygv** [comment]
The following content is generated by GPT, but the steps are very clear, I hope it is useful.
I think the most important content is `manifest`, we can imitate the previous `clash verge`.

To add the application "clash-nyanpasu" to the Scoop extras bucket, you would indeed need to create a new application manifest (a JSON file that describes how to install the application) and then submit it as a pull request to the Scoop extras repository.

Here are the steps you would generally follow:

1. **Fork the Scoop extras bucket repository:** Visit the [Scoop extras bucket](https://github.com/lukesampson/scoop-extras) on GitHub and click the "Fork" button.

2. **Clone your forked repository:** Use the command `git clone <URL of your forked repository>` to clone the repository to your local machine.

3. **Create a new JSON manifest for the application:** Go to the bucket directory (`cd scoop-extras/bucket`), and create a new text file in this directory with the name of your app and a `.json` extension (for example, `clash-nyanpasu.json`). In this file, you'll describe the application according to Scoop's application manifest format.

   Here's an example of what your `clash-nyanpasu.json` contents might look like:

   ```json
   {
     "version": "1.0.1",
     "description": "A Clash rule that...",
     "homepage": "https://github.com/keiko233/clash-nyanpasu",
     "license": "MIT",
     "url": "https://github.com/keiko233/clash-nyanpasu/archive/1.0.1.zip",
     "hash": "<fill in the correct SHA-256 hash of the zip here>",
     "bin": "clash-nyanpasu.exe",
     "checkver": "github",
     "autoupdate": {
       "url": "https://github.com/keiko233/clash-nyanpasu/archive/$version.zip"
     }
   }
    ```
  
   Make sure to replace the necessary fields with the correct information like the "version", "url", "hash" values. The "version", "url" and "hash" fields should correspond to the latest release of the application.
    Or you can ref https://github.com/ScoopInstaller/Extras/blob/master/bucket/clash-verge.json
   ```json
       {
        "version": "1.3.8",
        "description": "A Clash GUI based on Tauri.",
        "homepage": "https://github.com/zzzgydi/clash-verge",
        "license": "GPL-3.0-only",
        "notes": "zzzgydi/clash-verge has been archived, continue to use 'scoop install extras/clash-verge-rev' instead",
        "architecture": {
            "64bit": {
                "url": "https://github.com/zzzgydi/clash-verge/releases/download/v1.3.8/Clash.Verge_1.3.8_x64_portable.zip",
                "hash": "a7c296228e054930fabdbfd836279d0d3c413b1a08dcaa4604c540bf8905d529"
            }
        },
        "shortcuts": [
            [
                "Clash Verge.exe",
                "Clash Verge"
            ]
        ],
        "persist": ".config",
        "post_install": [
            "if (!(Test-Path \"$persist_dir\\.config\\PORTABLE\")) {",
            "    New-Item -Path \"$persist_dir\\.config\\PORTABLE\" -ItemType file | Out-Null",
            "}"
        ],
        "pre_uninstall": [
            "if (!(is_admin)) { error \"$app requires admin rights to $cmd\"; break }",
            "Start-Process \"$dir\\resources\\uninstall-service.exe\" -Wait -Verb 'RunAs' -WindowStyle 'Hidden'; Start-Sleep -Seconds 3"
        ],
        "checkver": "github",
        "autoupdate": {
            "architecture": {
                "64bit": {
                    "url": "https://github.com/zzzgydi/clash-verge/releases/download/v$version/Clash.Verge_$version_x64_portable.zip"
                }
            }
        }
    }
   ```

4. **Commit and push the changes:** Use `git add clash-nyanpasu.json`, `git commit -m "Add clash-nyanpasu"`, and `git push` commands to push your changes to your forked repository.

5. **Create a pull request:** Go to the GitHub page of your forked repository and create a new pull request. The owners of the Scoop extras bucket will then review your submission, and if everything is correct, they'll merge it into the Scoop extras bucket.

Please note that if the application binary releases are downloaded from GitHub, you could use `"checkver": "github"` and will be able to auto-generate hash by using `"autoupdate": { "hash": { "url": "$url" } }`.
The inclusion of your app in the bucket is subject to review and acceptance by the Scoop extras bucket maintainers.
